pkg/inmemorychannel/message_dispatcher.go

```diff
@@ -20,46 +20,45 @@ import (
 	"errors"
 	"time"
 
-	cloudevents "github.com/cloudevents/sdk-go/v1"
 	"go.uber.org/zap"
 
 	"knative.dev/eventing/pkg/channel/multichannelfanout"
 	"knative.dev/eventing/pkg/channel/swappable"
 	"knative.dev/eventing/pkg/kncloudevents"
 )
 
-type Dispatcher interface {
-	UpdateConfig(config *multichannelfanout.Config) error
+type MessageDispatcher interface {
+	UpdateConfig(context context.Context, config *multichannelfanout.Config) error
 }
 
-type InMemoryDispatcher struct {
-	handler      *swappable.Handler
-	ceClient     cloudevents.Client
-	writeTimeout time.Duration
-	logger       *zap.Logger
+type InMemoryMessageDispatcher struct {
+	handler              *swappable.MessageHandler
+	httpBindingsReceiver *kncloudevents.HttpMessageReceiver
+	writeTimeout         time.Duration
+	logger               *zap.Logger
 }
 
-type InMemoryDispatcherArgs struct {
+type InMemoryMessageDispatcherArgs struct {
 	Port         int
 	ReadTimeout  time.Duration
 	WriteTimeout time.Duration
-	Handler      *swappable.Handler
+	Handler      *swappable.MessageHandler
 	Logger       *zap.Logger
 }
 
-func (d *InMemoryDispatcher) UpdateConfig(config *multichannelfanout.Config) error {
-	return d.handler.UpdateConfig(config)
+func (d *InMemoryMessageDispatcher) UpdateConfig(context context.Context, config *multichannelfanout.Config) error {
```

Comment:
bit of a nit, typically we've been using:
ctx context.Context
not a big deal but for consistency might be nice :)

Reply:
done